systemverilog-plugin/UhdmAst.cc

```diff
@@ -1836,7 +1842,31 @@ void UhdmAst::simplify_parameter(AST::AstNode *parameter, AST::AstNode *module_n
         });
     }
     // first apply custom simplification step if needed
-    simplify(parameter, nullptr);
+    simplify(parameter, module_node);
+    // workaround for yosys sometimes not simplifying parameters children
+    // parameters can have 2 children:
+    // first child should be parameter value
+    // second child should be parameter range (optional)
+    log_assert(!parameter->children.empty());
+    simplify(parameter->children[0], parameter);
+    while (parameter->children[0]->simplify(true, false, false, 1, -1, false, false)) {
+    }
+    // follow id2ast as yosys doesn't do it by default
```

Comment:
Can you explain, why it is important to follow `id2ast` and pass it to _simplify_ here?

Reply:
When we are simplifying ``AST_IDENTIFIER`` yosys only links ``AST_IDENTIFIER`` to actual value (e.g. another ``AST_PARAMETER``)  by using id2ast field but doesn't try to simplify this linked node. To make sure yosys can simplify this  ``AST_IDENTIFIER`` we need to apply our custom simplification step to linked node.